functions/controllers/onboardingController.js

```diff
@@ -295,7 +303,29 @@ const setupUserProfile = functions.https.onCall(async (data, context) => {
         'Please input a valid full name.'
       );
     }
-
+    // Validate social media links
+    if (typeof socialLinks !== 'object' || socialLinks === null) {
+      logger.error(
+        'Error updating user profile, no social media links provided'
+      );
+      throw new functions.https.HttpsError(
+        'not-found',
+        'no social media links provided'
+      );
+    } else {
+      const links = Object.values(socialLinks);
+      // Checking that at least one link is filled in, throws an error if keys are provided by values are empty
+      const hasAtLeastOneLink = links.some((link) => link.trim() !== '');
+      if (!hasAtLeastOneLink) {
+        logger.error(
+          'Error updating user profile, at least one social media link is required'
+        );
+        throw new functions.https.HttpsError(
+          'internal',
+          'no social media links provided'
+        );
+      }
+    }
```

Comment:
I did not add validation to ensure no duplicate keys are provided e.g. 
-  facebook: link
- facebook: link
- twitter: link

As the preceding key-value pair is ignored automatically in javascript